drivers/flash/nrf_qspi_nor.c

```diff
@@ -556,11 +587,30 @@ static int qspi_nor_write(struct device *dev, off_t addr, const void *src,
 		return -EINVAL;
 	}
 
-	qspi_lock(dev);
-
-	nrfx_err_t res = nrfx_qspi_write(src, size, addr);
 
-	qspi_wait_for_completion(dev, res);
+	nrfx_err_t res;
+	if (IS_ENABLED(
```

Comment:
Nearly, I was rather thinking about something like this:
```
if (IS_ENABLED(CONFIG_NORDIC_QSPI_NOR_FLASH_ALLOW_STACK_USAGE_FOR_DATA_IN_FLASH) &&
    (((u32_t)src < CONFIG_SRAM_BASE_ADDRESS) && !(size < 4U)) {
...
} else {
....
}
```
This way you do not have to indent the body of if..else that deeply.

Reply:
Ah okei yeah that's better sorry for misunderstanding.

Reply:
No worries.